src/main/java/com/exactpro/th2/codec/grpc/ProtoDecoder.java

```diff
@@ -0,0 +1,106 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.exactpro.th2.codec.grpc;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+import com.exactpro.th2.common.grpc.Direction;
+import com.exactpro.th2.common.grpc.Message;
+import com.exactpro.th2.common.grpc.RawMessage;
+import com.exactpro.th2.common.grpc.RawMessageMetadata;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.DynamicMessage;
+import com.google.protobuf.InvalidProtocolBufferException;
+import org.apache.commons.io.FileUtils;
+
+public class ProtoDecoder {
+	public static final String GRPC_CALL = "GRPC_CALL";
+	private static final String TEMP_DIR = "gen/test/temp";
+
+	private final Map<String, Descriptors.ServiceDescriptor> serviceDescriptorMap = new HashMap<>();
+	
+	public ProtoDecoder(Path protosPath) throws IOException {
+		Collection<File> protos = FileUtils.listFiles(protosPath.toFile(), new String[]{"proto"}, true);
+		List<ProtoSchema> protoSchemas = new ProtobufParser(TEMP_DIR).parseProtosToSchemas(protosPath.toFile(), protos);
+		protoSchemas.forEach(schema -> serviceDescriptorMap.putAll(schema.getServices()));
+	}
+	
+	public Message decode(RawMessage message) throws InvalidProtocolBufferException, JsonProcessingException {
+		RawMessageMetadata metadata = message.getMetadata();
+		String grpcCall = metadata.getPropertiesOrThrow(GRPC_CALL);
+		Direction direction = metadata.getId().getDirection();
+		
+		Descriptors.Descriptor descriptor = getDescriptor(grpcCall, direction);
+		DynamicMessage dynamicMessage = DynamicMessage.newBuilder(descriptor)
+				.mergeFrom(message.getBody()).build();
+		return ProtoUtils.toMessage(dynamicMessage);
+	}
+	
+	private Descriptors.Descriptor getDescriptor(String grpcCallPath, Direction direction) {
+		GrpcCall grpcCall = new GrpcCall(grpcCallPath);
+		String serviceName = grpcCall.getService();
+
+		Descriptors.ServiceDescriptor serviceDesc = serviceDescriptorMap.get(serviceName);
+		if (serviceDesc == null)
+			throw new NoSuchElementException("There is no such service '" + serviceName + '\'');
+		
+		String methodName = grpcCall.getMethod();
+		Descriptors.MethodDescriptor methodDesc = serviceDesc.findMethodByName(methodName);
+		if (methodDesc == null)
+			throw new NoSuchElementException("There is no such method '" + methodName + "' for service '" + serviceName + '\'');
+
+		if (direction == Direction.FIRST)
+			return methodDesc.getInputType();
+		else
+			return methodDesc.getOutputType();
```

Comment:
```suggestion
		if (direction == Direction.FIRST) {
			return methodDesc.getInputType();
		} else {
			return methodDesc.getOutputType();
		}
```
Please use operator brackets to avoid the problem when code will be extended